invisible_cities/reco/deconv_functions.py

```diff
@@ -4,16 +4,78 @@
 from typing  import List
 from typing  import Tuple
 from typing  import Callable
+from typing import Optional
 
 from scipy                  import interpolate
 from scipy.signal           import fftconvolve
 from scipy.signal           import convolve
 from scipy.spatial.distance import cdist
+from scipy import ndimage as ndi
 
 from ..core .core_functions import shift_to_bin_centers
 from ..core .core_functions import in_range
 
 from .. types.symbols       import InterpolationMethod
+from .. types.symbols       import CutType
+
+
+
+import warnings
+
+
+def isolate_satellites(df : np.ndarray, dist : int = 2, size : int = 10) -> np.ndarray:
+    '''
+    An adaptation to the scikit-image (v0.24.0) function below:
+    https://github.com/scikit-image/scikit-image/blob/main/skimage/morphology/misc.py#L59-L151
+    
+    Identifies satellite energy depositions within deconvolution image by size
+    and proximity to other depositions.
+
+    In practice, input array is a set of boolean 'pixels' that it then categorises as 
+    satellite or non-satellite bundles based on given parameters.
+    
+    Parameters
+    ----------
+    df    : masked array containing pixels above energy cut
+    dist  : Maximum distance between two pixels to be considered part of the same deposit.
+    size  : Minimum number of pixels per deposit under which it will be labelled as a satellite deposit.
+
+    Returns
+    ----------
+    array : masked array of all satellite deposits
+
+    
+    '''
+    try:
+        array = data.copy().astype(bool)
+    except:
+        print("Please ensure the array passed through is boolean compatible.")
```

Comment:
You're right it should be an error.

> I believe the only realistic case where this would produce an exception is if the array is not numeric, right?

It can occur if somehow the array passed in non-boolean, rather than non-numeric. The array needs to be boolean for the satellite killer to work as intended.

Reply:
As far as I know, numeric arrays can always be cast to boolean.

Reply:
That's true, my worry was that if someone uses the function but passes through the non-masked array (so the raw deconvolved data), it would interpret all numbers > 0 as true, which would cause the output to end up very strange.

A better solution may be to pull the e_cut mask production from `richardson_lucy()` into `isolate_satellites()`, do you think that would be better?

Reply:
> if someone uses the function but passes through the non-masked array (so the raw deconvolved data), it would interpret all numbers > 0 as true

But this would not raise an exception, would it?

> A better solution may be to pull the e_cut mask production from richardson_lucy() into isolate_satellites(), do you think that would be better?

This sounds more appropriate, provided this mask is not used anywhere else. Perhaps a function `remove_satellites` that combines masking, isolating, and removing would be ideal if this is indeed the workflow.

Reply:
> But this would not raise an exception, would it?

Ah, yes you're right. How silly of me.

> Perhaps a function remove_satellites that combines masking, isolating, and removing would be ideal if this is indeed the workflow.

I think that would make sense.

Reply:
This part of the code is gone now.